doc/user/layouts/shortcodes/views-indexes/index-considerations.html

```diff
@@ -0,0 +1,13 @@
+Before creating an index, consider the following:
+
+- If you have created various views to act as the building blocks for a view
+  from which you will serve the results, creating an index **only** on the
+  serving view may be sufficient depending upon your query patterns and usage.
+
+- In Materialize, underlying data structures that are being maintained for an
+  index (or a materialized view) can be reused by other views/queries. As such,
+  an existing index may be sufficient to support your queries.
```

Comment:
```suggestion
- Materialize can reuse indexes across queries that concurrently access the same
  data in memory, which reduces redundancy and resource utilization per query.
  In particular, this means that joins don’t need to store data in memory
  multiple times.
```

Reply:
Useful blogposts to fish for information and best practices about indexes:

* https://materialize.com/blog/power-user/
* https://materialize.com/blog/maintaining-joins-using-few-resources/
* https://materialize.com/blog/delta-joins/
* https://materialize.com/blog/indexes-a-silent-frenemy/